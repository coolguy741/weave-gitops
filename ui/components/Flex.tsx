import * as React from "react";
import styled from "styled-components";

type Props = {
  className?: string;
  column?: boolean;
  align?: boolean;
  height?: string;
  between?: boolean;
  center?: boolean;
  wide?: boolean;
  wrap?: boolean;
  shadow?: boolean;
  onMouseEnter?: React.ReactEventHandler;
  onMouseLeave?: React.ReactEventHandler;
};

const Styled = (component) => styled(component)`
  display: flex;
  flex-direction: ${({ column }) => (column ? "column" : "row")};
  align-items: ${({ align }) => (align ? "center" : "start")};
  ${({ tall }) => tall && `height: 100%`};
  ${({ wide }) => wide && "width: 100%"};
  ${({ wrap }) => wrap && "flex-wrap: wrap"};
  ${({ start }) => start && "justify-content: flex-start"};
  ${({ end }) => end && "justify-content: flex-end"};
  ${({ between }) => between && "justify-content: space-between"};
  ${({ center }) => center && "justify-content: center"};
  ${({ shadow }) => shadow && "box-shadow: 5px 10px 50px 3px #0000001a"};
`;

class Flex extends React.PureComponent<Props> {
  render() {
    const { className, children, onMouseEnter, onMouseLeave } = this.props;
    return (
      <div
        className={className}
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
      >
        {children}
      </div>
    );
  }
}

export const MessageFlex = styled(Styled(Flex))`
  box-sizing: border-box;
  width: 560px;
  padding: ${({ theme }) => theme.spacing.medium}
    ${({ theme }) => theme.spacing.xl} ${({ theme }) => theme.spacing.xxl};
  border-radius: 10px;
  background-color: #ffffffd9;
  color: ${({ theme }) => theme.colors.neutral30};
`;

MessageFlex.defaultProps = {
  shadow: true,
  align: true,
  column: true,
};

export default Styled(Flex);
