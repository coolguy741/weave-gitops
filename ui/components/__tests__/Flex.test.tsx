import "jest-styled-components";
import React from "react";
import renderer from "react-test-renderer";
import Flex from "../Flex";

describe("Flex", () => {
  describe("snapshots", () => {
    it("wide", () => {
      const tree = renderer.create(<Flex wide>My Text</Flex>).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("center", () => {
      const tree = renderer.create(<Flex center>My Text</Flex>).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("align", () => {
      const tree = renderer.create(<Flex align>My Text</Flex>).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("wide center align", () => {
      const tree = renderer
        .create(
          <Flex wide center align>
            My Text
          </Flex>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
