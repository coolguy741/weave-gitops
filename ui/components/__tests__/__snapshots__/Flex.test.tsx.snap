// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex snapshots align 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
>
  My Text
</div>
`;

exports[`Flex snapshots center 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
>
  My Text
</div>
`;

exports[`Flex snapshots wide 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  width: 100%;
}

<div
  className="c0"
>
  My Text
</div>
`;

exports[`Flex snapshots wide center align 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
>
  My Text
</div>
`;
